pkg/resource/deploy/step.go

```diff
@@ -356,15 +356,23 @@ func isDeletedWith(with resource.URN, otherDeletions map[resource.URN]bool) bool
 	return r
 }
 
+type deleteProtectedError struct {
+	urn resource.URN
+}
+
+func (d deleteProtectedError) Error() string {
+	return fmt.Sprintf("resource %[1]q is marked for protection\n"+
+		"so it cannot be deleted. To unprotect the resource, "+
+		"either remove the `protect` flag from the resource in your Pulumi "+
+		"program and run `pulumi up` or use the command:\n"+
```

Comment:
smallest of nits:
comma between the two options
```suggestion
		"program and run `pulumi up`, or use the command:\n"+
```